meshtastic/mesh_interface.py

```diff
@@ -314,19 +314,33 @@ def getTimeAgo(ts) -> Optional[str]:
         return table
 
     def getNode(
-        self, nodeId: str, requestChannels: bool = True
+        self, nodeId: str, requestChannels: bool = True, requestChannelRetries: int = 3, timeout: int = 300
     ) -> meshtastic.node.Node:
         """Return a node object which contains device settings and channel info"""
         if nodeId in (LOCAL_ADDR, BROADCAST_ADDR):
             return self.localNode
         else:
-            n = meshtastic.node.Node(self, nodeId)
+            n = meshtastic.node.Node(self, nodeId, timeout=timeout)
             # Only request device settings and channel info when necessary
             if requestChannels:
                 logging.debug("About to requestChannels")
                 n.requestChannels()
-                if not n.waitForConfig():
-                    our_exit("Error: Timed out waiting for channels")
+                retries_left = requestChannelRetries
+                last_index: int = 0
+                while retries_left > 0:
+                    retries_left -= 1
+                    if not n.waitForConfig():
+                        new_index: int = len(n.partialChannels)
+                        # each time we get a new channel, reset the counter
+                        if new_index != last_index:
+                            retries_left = requestChannelRetries - 1
+                        if retries_left <= 0:
+                            our_exit(f"Error: Timed out waiting for channels, giving up")
+                        print("Timed out trying to retrieve channel info, retrying")
+                        n.requestChannels(startingIndex=len(n.partialChannels))
```

Comment:
I think the CI failures here for the types are correct in that `partialChannels` isn't guaranteed to be a list. I think you could do:

```python
new_index: int = len(n.partialChannels) if n.partialChannels else 0
```
And, you may as well change the `requestChannels` call on line 340 to just reuse the `new_index` variable as well.